src/managers/GuildInviteManager.js

```diff
@@ -0,0 +1,185 @@
+const Collection = require('../util/Collection');
+const DataResolver = require('../util/DataResolver');
+
+/**
+ * Manages API methods for GuildInvites and stores their cache.
+ * @extends {BaseManager}
+ */
+class GuildInviteManager extends BaseManager {
+  constructor(guild, iterable) {
+    super(guild.client, iterable, Invite);
+
+    /**
+     * The guild this Manager belongs to
+     * @type {Guild}
+     */
+    this.guild = guild;
+  }
+
+  /**
+   * The cache of this Manager
+   * @type {Collection<Snowflake, Invite>}
+   * @name GuildInviteManager#cache
+   */
+
+  add(data, cache) {
+    return super.add(data, cache, { id: data.code, extras: [this.guild] });
+  }
+
+  /**
+   * Data that resolves to give a Invite object. This can be:
+   * * An invite code
+   * * An invite URL
+   * @typedef {string} InviteResolvable
+   */
+
+  /**
+   * Resolves a InviteResolvable to a Invite object.
+   * @method resolve
+   * @memberof GuildInviteManager
+   * @instance
+   * @param {InviteResolvable} invite The invite resolvable to resolve
+   * @returns {?Invite}
+   */
+
+  /**
+   * Resolves a InviteResolvable to a invite code string.
+   * @method resolveID
+   * @memberof GuildInviteManager
+   * @instance
+   * @param {InviteResolvable} invite The invite resolvable to resolve
+   * @returns {?string}
+   */
+
+  /**
+   * Options used to fetch a single invite from a guild.
+   * @typedef {Object} FetchInviteOptions
+   * @property {InviteResolvable} code The invite to fetch
+   * @property {boolean} [cache=true] Whether or not to cache the fetched invite
+   * @property {boolean} [force=false] Whether to skip the cache check and request the API
+   */
+
+  /**
+   * Options used to fetch all invites from a guild.
+   * @typedef {Object} FetchInvitesOptions
+   * @property {boolean} cache Whether or not to cache the fetched invites
+   */
+
+  /**
+   * Fetches invite(s) from Discord.
+   * @param {InviteResolvable|FetchInviteOptions|FetchInvitesOptions} [options] Options for fetching guild invite(s)
+   * @returns {Promise<Invite>|Promise<Collection<Snowflake, Invite>>}
+   * @example
+   * // Fetch all invites from a guild
+   * guild.invites.fetch()
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch all invites from a guild without caching
+   * guild.invites.fetch({ cache: false })
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch a single invite
+   * guild.invites.fetch('bRCvFy9')
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch a single invite without checking cache
+   * guild.invites.fetch({ code: 'bRCvFy9', force: true })
+   *   .then(console.log)
+   *   .catch(console.error)
+   * @example
+   * // Fetch a single invite without caching
+   * guild.invites.fetch({ code: 'bRCvFy9', cache: false })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  fetch(options) {
```

Comment:
This should use async/await.

Reply:
There isn't any await in this function, why should it be needed?

Reply:
The function is full of promise rejections that happen by using `return Promise.reject(new <>Error())` and that could be a lot cleaner by writing `throw new <>Error()` instead.

Also, in case something ever throws a synchronous error, such as reading a property of undefined, it'd be a promise rejection instead of a regular exception.

Reply:
ESLint is not likely to be against if there is any await?

Reply:
It shouldn't if there are `throw`s, but who knows.

Reply:
https://eslint.org/docs/rules/require-await#when-not-to-use-it